lifting-state-up/passing-down-event-handlers.md

```diff
@@ -80,43 +158,128 @@ This function should go inside the `App` component, after `studentData` is defin
 
 <br/>
 
-Because React wouldn't notice the change! For container types, like arrays and objects, if we update an inner value (an item in an array, or a field in an object) and then call the relevant set function, React will see that the reference to the object currently in state is the same reference as the object being passed into the set function. For performance reasons, it assumes that if the same reference is passed in to a set function, it _is_ the same object. It will not look within the array or object for changes!
+<details>
+<summary>Click here to see an explanation</summary>
+
+<br />
+
+We need to use a whole new array because otherwise React wouldn't notice the change! For container types, like arrays and objects, if we update an inner value (an item in an array, or a field in an object) and then call the relevant set function, React will see that the reference to the object currently in state is the same reference as the object being passed into the set function. For performance reasons, it assumes that if the same reference is passed in to a set function, it _is_ the same object. It will not look within the array or object for changes!
 
 <br/>
 
 So to get React to notice that a value in our array or object has changed, we need to make a new outer reference, and copy the existing values into it. Then when we call a set function, it will see the change and trigger a re-render.
 
+</details>
+
+### !end-callout
+
+<!-- available callout types: info, success, warning, danger, secondary, star  -->
+### !callout-info
+
+## Refactoring Opportunity: Using Function-Passing State Updates
+
+We can update arrays using the function-passing style of calling state update functions. Recall that the function we pass to the updater will be called with the current value of the state (our students array) as its first parameter. Our function should return a new value (a new students array) that React will use to update the related piece of state.
+
+<br />
+
+Think about what changes we would need to make to `toggleStudentPresence`, then expand the section below to see a possible solution.
+
+<br />
+
+<details>
+<summary>Click here to see a possible solution</summary>
+
+```js
+  const toggleStudentPresence = (studentId) => {
+    setStudentData(students => {
+      return students.map(student => {
+        if (student.id === studentId) {
+          return { ...student, isPresentData: !student.isPresentData };
+        } else {
+          return student;
+        }
+      });
+    });
+  };
+```
+
+<br />
+
+While the behavior of both this version and the previous version are the same at present. If we extended the application to work with data from a server, this version would be more robust. It would ensure that the data we are updating is the most recent version of the data, rather than the version that was in state when the function was defined, avoiding potential bugs.
```

Comment:
```suggestion
While the behavior of both this version and the previous version are the same at present if we extended the application to work with data from a server, this version would be more robust. It would ensure that the data we are updating is the most recent version of the data, rather than the version that was in state when the function was defined, avoiding potential bugs.
```

Reply:
```suggestion
While the behavior of both this version and the previous version are the same at present, if we extended the application to work with data from a server, this version would be more robust. It would ensure that the data we are updating is the most recent version of the data, rather than the version that was in state when the function was defined, avoiding potential bugs.
```